File: spring-boot-rabbitmq-demo/src/main/java/com/sivalabs/demo/ProductRepository.java
package com.sivalabs.demo;

import java.math.BigDecimal;
import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Modifying;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;

public interface ProductRepository extends JpaRepository<Product, Long> {

    @Query("update Product p set p.price = :price where p.code = :productCode")
    @Modifying
    void updateProductPrice(@Param("productCode") String productCode, @Param("price") BigDecimal price);

    Optional<Product> findByCode(String code);
}

File: spring-boot-kotlin-demo/src/main/kotlin/com/sivalabs/demo/ProductRepository.kt
package com.sivalabs.demo

import org.springframework.jdbc.core.JdbcTemplate
import org.springframework.stereotype.Repository

@Repository
class ProductRepository(
    private val jdbcTemplate: JdbcTemplate,
) {
    fun getAll(): List<Product> =
        jdbcTemplate.query(
            "select * from products",
        ) { rs, _ ->
            Product(
                rs.getLong("id"),
                rs.getString("code"),
                rs.getString("name"),
                rs.getBigDecimal("price"),
            )
        }
}
